### ISSUE TITLE
Dotfile search with ignore filtering out all responses, never emits 'end'

### ISSUE THREAD

**Issue by thirdcreed** [isaacs/node-glob#232]
```javascript
require("glob").Glob("**/home/thirdcreed/Projects/glob-stream/test/fixtures/*swag**", {
  ignore: [
    "/home/thirdcreed/Projects/glob-stream/test/fixtures/**"
  ],
  dot:true,
  cwd: "/home/thirdcreed/Projects/glob-stream/test",
  cwdbase:false,
  nonull:false
}).on("end",function(x){
  console.log("end stream emitted",x);
})
```

emits nothing...I can trace it all the way to inside the _finish block of node_glob's *glob.js*, and I console.log it on both sides of the emit, it logs on both sides of the emit, but it never seems to fire.

BUT

```javascript
require("glob").Glob("**/home/thirdcreed/Projects/glob-stream/test/fixtures/*swag**", {
  ignore:[
    "**/home/thirdcreed/Projects/glob-stream/test/DOESNOTCOVER/**"
  ],
  dot:true,
  cwd: "/home/thirdcreed/Projects/glob-stream/test",
  cwdbase:false,
  nonull:false
}).on("end",function(x){
  console.log("end stream emitted",x);
})
```

emits with no problems.

The first one has an ignore that filters all files out.

The second has an ignore that doesn't filter at all.

**Comment by thirdcreed** [comment]
Then somehow this emits 

node -p 'new require("glob").Glob("**/home/thirdcreed/Projects/glob-stream/test/fixtures/*swag**",{ignore:["**/home/thirdcreed/Projects/glob-stream/test/fixtures/*swag**"],dot:true,cwd: "/home/thirdcreed/Projects/glob-stream/test",cwdbase:false,nonull:false}).on("end",function(x){console.log("end stream emitted",x);})'

Any insight from anyone would be much appreciated, this is really confusing me.

**Comment by thirdcreed** [comment]
It's a race condition.  The glob object isn't instantiated when the event is emitted.  The event isn't registered because there's no listeners attached.

**Comment by isaacs** [comment]
Indeed!  Fixed on 7.0.1.
